`test_keeper_broken_logs/test.py::test_single_node_broken_log` is flaky

https://s3.amazonaws.com/clickhouse-test-reports/56825/ef17d972ab64c90ff72a7c6c2beba61b522d6fcf/integration_tests__release__[4_4].html

CI DB: [link](https://play.clickhouse.com/play?user=play#CldJVEgKICAgICd0ZXN0X3NpbmdsZV9ub2RlX2Jyb2tlbl9sb2cnIEFTIG5hbWVfc3Vic3RyLAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cywKICAgICgnU3RhdGVsZXNzIHRlc3RzIChhc2FuKScsICdTdGF0ZWxlc3MgdGVzdHMgKGFkZHJlc3MpJywgJ1N0YXRlbGVzcyB0ZXN0cyAoYWRkcmVzcywgYWN0aW9ucyknKSBBUyBiYWNrcG9ydF9hbmRfcmVsZWFzZV9zcGVjaWZpY19jaGVja3MKU0VMRUNUCiAgICB0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIEFTIGQsCiAgICBjb3VudCgpLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpCkZST00gY2hlY2tzCldIRVJFICgobm93KCkgLSB0b0ludGVydmFsRGF5KGludGVydmFsX2RheXMpKSA8PSBjaGVja19zdGFydF90aW1lKSBBTkQgKHB1bGxfcmVxdWVzdF9udW1iZXIgTk9UIElOICgKICAgIFNFTEVDVCBwdWxsX3JlcXVlc3RfbnVtYmVyIEFTIHBybgogICAgRlJPTSBjaGVja3MKICAgIFdIRVJFIChwcm4gIT0gMCkgQU5EICgobm93KCkgLSB0b0ludGVydmFsRGF5KGludGVydmFsX2RheXMpKSA8PSBjaGVja19zdGFydF90aW1lKSBBTkQgKGNoZWNrX25hbWUgSU4gKGJhY2twb3J0X2FuZF9yZWxlYXNlX3NwZWNpZmljX2NoZWNrcykpCikpIEFORCAocG9zaXRpb24odGVzdF9uYW1lLCBuYW1lX3N1YnN0cikgPiAwKSBBTkQgKHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicsICdGTEFLWScpKQpHUk9VUCBCWSBkCk9SREVSIEJZIGQgREVTQwo=)

cc: @antonio2368

It's flaky again: https://s3.amazonaws.com/clickhouse-test-reports/66532/795abfcb0a78202967d915bcac707ca9cb6b340c/integration_tests__tsan__[6_6].html

```
>           assert (
                node1.exec_in_container(["ls", "/var/lib/clickhouse/coordination/log"])
                == "changelog_1_100000.bin\nchangelog_14_100013.bin\n"
            )
E           AssertionError: assert 'changelog_1_100000.bin\nchangelog_16_100015.bin\n' == 'changelog_1_100000.bin\nchangelog_14_100013.bin\n'
E               changelog_1_100000.bin
E             - changelog_14_100013.bin
E             ?            ^      ^
E             + changelog_16_100015.bin
E             ?            ^      ^
```

--- a/tests/integration/test_keeper_broken_logs/test.py
+++ b/tests/integration/test_keeper_broken_logs/test.py
@@ -5,6 +5,8 @@ import pytest
 import helpers.keeper_utils as keeper_utils
 from helpers.cluster import ClickHouseCluster
 
+from multiprocessing.dummy import Pool
+
 cluster = ClickHouseCluster(__file__)
 node1 = cluster.add_instance(
     "node1",
@@ -52,14 +54,33 @@ def get_fake_zk(nodename, timeout=30.0):
     return _fake_zk_instance
 
 
+def start_clickhouse(node):
+    node.start_clickhouse()
+
+
+def clean_start():
+    nodes = [node1, node2, node3]
+    for node in nodes:
+        node.stop_clickhouse()
+
+    p = Pool(3)
+    waiters = []
+    for node in nodes:
+        node.exec_in_container(["rm", "-rf", "/var/lib/clickhouse/coordination/log"])
+        node.exec_in_container(
+            ["rm", "-rf", "/var/lib/clickhouse/coordination/snapshots"]
+        )
+        waiters.append(p.apply_async(start_clickhouse, (node,)))
+
+    for waiter in waiters:
+        waiter.wait()
+
+
 def test_single_node_broken_log(started_cluster):
+    clean_start()
     try:
         wait_nodes()
         node1_conn = get_fake_zk("node1")
-
-        # Cleanup
-        if node1_conn.exists("/test_broken_log") != None:
-            node1_conn.delete("/test_broken_log")
 
         node1_conn.create("/test_broken_log")
         for _ in range(10):
@@ -110,10 +131,12 @@ def test_single_node_broken_log(started_cluster):
         verify_nodes(node3_conn)
         assert node3_conn.get("/test_broken_log_final_node")[0] == b"somedata1"
 
-        assert (
+        node1_logs = (
             node1.exec_in_container(["ls", "/var/lib/clickhouse/coordination/log"])
-            == "changelog_1_100000.bin\nchangelog_14_100013.bin\n"
+            .strip()
+            .split("\n")
         )
+        assert len(node1_logs) == 2 and node1_logs[0] == "changelog_1_100000.bin"
         assert (
             node2.exec_in_container(["ls", "/var/lib/clickhouse/coordination/log"])
             == "changelog_1_100000.bin\n"

--- a/tests/integration/test_keeper_broken_logs/test.py
+++ b/tests/integration/test_keeper_broken_logs/test.py
@@ -1,11 +1,10 @@
 import time
+from multiprocessing.dummy import Pool
 
 import pytest
 
 import helpers.keeper_utils as keeper_utils
 from helpers.cluster import ClickHouseCluster
-
-from multiprocessing.dummy import Pool
 
 cluster = ClickHouseCluster(__file__)
 node1 = cluster.add_instance(

